daemon/anonymous-statistics.sh

```diff
@@ -0,0 +1,157 @@
+# -------------------------------------------------------------------------------------------------
+# check opt-out
+
+OPTOUT=
+if [ -z "${NETDATA_CONFIG_DIR}" ]; then
+    for dir in "$(netdata -W get global 'config directory' '/etc/netdata' 2>/dev/null)" "/etc/netdata" "/opt/netdata/etc/netdata"; do
+        if [ -f "${dir}/.opt-out-from-anonymous-statistics" ]; then
+            OPTOUT=all
+            break
+        fi
+    done
+else
+    if [ -f "${NETDATA_CONFIG_DIR}/.opt-out-from-anonymous-statistics" ]; then
+        OPTOUT=all
+    fi
+fi
+
+if [ "${OPTOUT}" = "all" ] ; then
+    exit 0
+fi
+
+# -------------------------------------------------------------------------------------------------
+# detect the operating system
+
+OS_DETECTION=
+NAME=
+VERSION=
+VERSION_ID=
+ID=
+ID_LIKE=
+
+if [ -f "/etc/os-release" ]; then
+    OS_DETECTION="/etc/os-release"
+    eval "$(grep -E "^(NAME|ID|ID_LIKE|VERSION|VERSION_ID)=" </etc/os-release)"
+elif [ -f "/etc/lsb-release" ]; then
+    OS_DETECTION="/etc/lsb-release"
+    DISTRIB_ID=
+    DISTRIB_RELEASE=
+    DISTRIB_CODENAME=
+    eval "$(grep -E "^(DISTRIB_ID|DISTRIB_RELEASE|DISTRIB_CODENAME)=" </etc/lsb-release)"
+    NAME="${DISTRIB_ID}"
+    VERSION="${DISTRIB_RELEASE}"
+    ID="${DISTRIB_CODENAME}"
+elif [ ! -z "$(command -v lsb_release 2>/dev/null)" ]; then
+    OS_DETECTION="lsb_release"
+    NAME="$(lsb_release -i 2>/dev/null | sed "s/[[:space:]]*:[[:space:]]*/:/g" | cut -d ':' -f 2)"
+    VERSION_ID="$(lsb_release -f 2>/dev/null | sed "s/[[:space:]]*:[[:space:]]*/:/g" | cut -d ':' -f 2)"
+    VERSION="$(lsb_release -c 2>/dev/null | sed "s/[[:space:]]*:[[:space:]]*/:/g" | cut -d ':' -f 2)"
+fi
+
+# -------------------------------------------------------------------------------------------------
+# detect the kernel
+
+KERNEL_NAME="$(uname -s)"
+KERNEL_VERSION="$(uname -r)"
+ARCHITECTURE="$(uname -m)"
+
+# -------------------------------------------------------------------------------------------------
+# detect the virtualization
+
+VIRTUALIZATION="unknown"
+VIRT_DETECTION="none"
+
+if [ ! -z "$(command -v systemd-detect-virt 2>/dev/null)" ]
+then
+    VIRTUALIZATION="$(systemd-detect-virt)"
+    VIRT_DETECTION="systemd-detect-virt"
+else
+    # /proc/1/sched exposes the host's pid of our init !
+    # http://stackoverflow.com/a/37016302
+    pid=$( head -n 1 </proc/1/sched 2>/dev/null | { IFS='(),#:' read -r _unused pid _unused _unused; echo $pid; } )
+    if [ ! -z "${pid}" ]; then
+        pid=$(( pid + 0 ))
+        [ ${pid} -ne 1 ] &&
+            VIRTUALIZATION="container" &&
+            VIRT_DETECTION="pid"
+    fi
+
+    # ubuntu and debian supply /bin/running-in-container
+    # https://www.apt-browse.org/browse/ubuntu/trusty/main/i386/upstart/1.12.1-0ubuntu4/file/bin/running-in-container
+    [ -x "/bin/running-in-container" ] &&
+        "/bin/running-in-container" >/dev/null 2>&1 &&
+        VIRTUALIZATION="container" &&
+        VIRT_DETECTION="/bin/running-in-container"
```

Comment:
Whole script `/bin/running-in-container` just checks for the content of `/run/container_type` which in turn should contain a containerization type (our variable called VIRTUALIZATION). So maybe instead of checking for this file, we should just `cat /run/container_type` and assign it to `$VIRTUALIZATION`?

Source: I went to a link from a comment in L98.

Also to be fair, this file needs to have upstart installed, which limits scope of this check only to older ubuntu distributions running inside lxc (docker images shouldn't have init system).

Reply:
We assign it to CONTAINER now and we do run other checks as well. So I'l like to leave it as is to finish.